src/XIVLauncher.Common.Unix/Compatibility/UnixHelpers.cs

```diff
@@ -0,0 +1,96 @@
+using XIVLauncher.Common.Util;
+
+namespace XIVLauncher.Common.Unix.Compatibility;
+
+public enum DistroPackage
+{
+    ubuntu,
+
+    fedora,
+
+    arch,
+
+    none,
+}
+
+public static class UnixHelpers
+{
+    public static async Task InstallDxvk(DirectoryInfo prefix, DirectoryInfo installDirectory, string folder, string downloadUrl)
+    {
+        var dxvkPath = Path.Combine(installDirectory.FullName, folder, "x64");
+        if (!Directory.Exists(dxvkPath))
+        {
+            Log.Information("DXVK does not exist, downloading");
+            await DownloadTool(installDirectory, downloadUrl).ConfigureAwait(false);
+        }
+
+        var system32 = Path.Combine(prefix.FullName, "drive_c", "windows", "system32");
+        var files = Directory.GetFiles(dxvkPath);
+
+        foreach (string fileName in files)
+        {
+            File.Copy(fileName, Path.Combine(system32, Path.GetFileName(fileName)), true);
+        }
+
+        // 32-bit files for Directx9.
+        var dxvkPath32 = Path.Combine(installDirectory.FullName, folder, "x32");
+        if (Directory.Exists(dxvkPath32))
+        {
+            var syswow64 = Path.Combine(prefix.FullName, "drive_c", "windows", "syswow64");
+            files = Directory.GetFiles(dxvkPath32);
+
+            foreach (string fileName in files)
+            {
+                File.Copy(fileName, Path.Combine(syswow64, Path.GetFileName(fileName)), true);
+            }
+        }   
+    }
+
+    public static async Task DownloadWine(DirectoryInfo installDirectory, string folder, string downloadUrl)
+    {
+        if (!Directory.Exists(Path.Combine(installDirectory.FullName, folder)))
+            await DownloadTool(installDirectory, downloadUrl).ConfigureAwait(false);
+    }
+
+    private static async Task DownloadTool(DirectoryInfo installDirectory, string downloadUrl)
+    {
+        using var client = new HttpClient();
+        var tempPath = Path.GetTempFileName();
+
+        File.WriteAllBytes(tempPath, await client.GetByteArrayAsync(downloadUrl));
+        PlatformHelpers.Untar(tempPath, installDirectory.FullName);
+
+        File.Delete(tempPath);
+    }
+    public static bool DxvkHudStringIsValid(string customHud)
+    {
+        var ALLOWED_CHARS = "^[0-9a-zA-Z,=.]+$";
+        var ALLOWED_WORDS = "^(?:devinfo|fps|frametimes|submissions|drawcalls|pipelines|descriptors|memory|gpuload|version|api|cs|compiler|samplers|scale=(?:[0-9])*(?:.(?:[0-9])+)?)$";
+
+        if (string.IsNullOrWhiteSpace(customHud)) return false;
+        if (customHud == "full") return true;
+        if (customHud == "1") return true;
+        if (!Regex.IsMatch(customHud, ALLOWED_CHARS)) return false;
+
+        string[] hudvars = customHud.Split(",");
+
+        return hudvars.All(hudvar => Regex.IsMatch(hudvar, ALLOWED_WORDS));        
+    }
+
+    public static bool MangoHudIsInstalled()
+    {
+        var usrLib = Path.Combine("/usr", "lib", "mangohud", "libMangoHud.so"); // fedora uses this
```

Comment:
`Path.Combine("/", "usr", "lib", "mangohud", "libMangoHud.so");` is a bit more elegant

Reply:
Didn't realize that was how it worked for unix paths. Also didn't realize you could go over four strings without making a string array. That is a bit nicer, certainly. I also went and fixed acouple of things in the XL.Core side for the same thing.